homeassistant/components/lookin/media_player.py

```diff
@@ -0,0 +1,224 @@
+        )
+        await coordinator.async_refresh()
+        device: Remote = coordinator.data
+
+        entities.append(
+            LookinMedia(
+                uuid=uuid,
+                device=device,
+                lookin_data=lookin_data,
+                device_class=_TYPE_TO_DEVICE_CLASS[remote["Type"]],
+                coordinator=coordinator,
+            )
+        )
+
+    async_add_entities(entities)
+
+
+class LookinMedia(LookinPowerEntity, MediaPlayerEntity):
+    """A lookin media player."""
+
+    _attr_should_poll = False
+
+    def __init__(
+        self,
+        uuid: str,
+        device: Remote,
+        lookin_data: LookinData,
+        device_class: str,
+        coordinator: DataUpdateCoordinator,
+    ) -> None:
+        """Init the lookin media player."""
+        self._attr_device_class = device_class
+        self._attr_supported_features: int = 0
+        self._state: str | None = None
+        self._is_muted: bool = False
+        super().__init__(coordinator, uuid, device, lookin_data)
+        for function_name, feature in _FUNCTION_NAME_TO_FEATURE.items():
+            if function_name in self._function_names:
+                self._attr_supported_features |= feature
+        self._async_update_from_data()
+
+    @property
+    def _remote(self) -> Remote:
+        return cast(Remote, self.coordinator.data)
+
+    @property
+    def state(self) -> str | None:
+        """State of the player."""
+        return self._state
+
+    async def async_volume_up(self) -> None:
+        """Turn volume up for media player."""
+        await self._async_send_command("volup")
+
+    async def async_volume_down(self) -> None:
+        """Turn volume down for media player."""
+        await self._async_send_command("voldown")
+
+    async def async_media_previous_track(self) -> None:
+        """Send previous track command."""
+        await self._async_send_command("chdown")
+
+    async def async_media_next_track(self) -> None:
+        """Send next track command."""
+        await self._async_send_command("chup")
+
+    async def async_mute_volume(self, mute: bool) -> None:
+        """Mute the volume."""
+        await self._async_send_command("mute")
+        self._is_muted = not self._is_muted
+
+    async def async_turn_off(self) -> None:
+        """Turn the media player off."""
+        await self._async_send_command(self._power_off_command)
+        self._state = STATE_STANDBY
+
+    async def async_turn_on(self) -> None:
+        """Turn the media player on."""
+        await self._async_send_command(self._power_on_command)
+        self._state = STATE_ON
+
+    @property
+    def name(self) -> str:
+        """Return the name of the entity."""
+        name: str = self._remote.name
+        return name
+
+    @property
+    def is_volume_muted(self) -> bool | None:
+        """Boolean if volume is currently muted."""
+        return self._is_muted
```

Comment:
```suggestion
```

Drop this in favor of https://github.com/home-assistant/core/blob/dev/homeassistant/components/media_player/__init__.py#L458